plugins/builds/index.js

```diff
@@ -961,23 +967,36 @@ const buildsPlugin = {
 
             for (const pid of Object.keys(pipelineJoinData)) {
                 // typecast pid to number
-                if (+pid === current.pipeline.id) {
+                let triggerCurrentPipelineAsExternal = false;
+                const isCurrentPipeline = +pid === current.pipeline.id;
+
+                if (isCurrentPipeline) {
                     for (const nextJobName of Object.keys(pipelineJoinData[pid].jobs)) {
                         try {
-                            const resource = `pipeline:${current.pipeline.id}:event:${current.event.id}`;
-                            const lock = await locker.lock(resource);
+                            const forceExternalTrigger = pipelineJoinData[pid].jobs[nextJobName].isExternal;
+
+                            triggerCurrentPipelineAsExternal = triggerCurrentPipelineAsExternal || forceExternalTrigger;
+                            if (!forceExternalTrigger) {
```

Comment:
```suggestion
                            if (!triggerCurrentPipelineAsExternal) {
```

Reply:
no this won't be correct, there could be mix of jobs, some in external format and some not. we are looking specifically at current job